leaflet.filelayer.js

```diff
@@ -82,13 +88,81 @@ var FileLoader = L.Class.extend({
+     */
+    _convertCsvToGeoJSON: function(content){
+        try {
+            if(!this.options.firstLineTitles){
+                throw new Error('The file CSV must have the Headers');
+            }
+            var geojson;
+            /*
+             * for this function i used the Papa Parser of mholt
+             * (https://github.com/mholt/PapaParse)
+             */
+            geojson = Papa.parse(content,{header: this.options.firstLineTitles}); //convert csv to json.
+            geojson = this._addFeatureToJson(geojson);
+            return this._loadGeoJSON(geojson);
+        }catch(e){alert(e.message);}
+    },
+
+    _addFeatureToJson: function(papaJson){
+        papaJson["type"]="FeatureCollection";
+        papaJson["features"]=[];
+        var title = papaJson.meta.fields;
+        for (var num_linea = 0; num_linea < papaJson.data.length - 1; num_linea++) { //  var depth = papaJson.data.length - 1;
+            var obj = papaJson.data[num_linea]; //single element of papa parse json object
+            var fields = title.toString().trim().split(",");
+            var lng = parseFloat(obj[this.options.longitudeTitle]);
+            var lat = parseFloat(obj[this.options.latitudeTitle]);
+            if (fields.length+"=="+title.length && lng<180 && lng>-180 && lat<90 && lat>-90) {
+                var feature = {};
+                feature["type"]="Feature";
+                feature["geometry"]={};
+                feature["properties"]={};
+                feature["geometry"]["type"]="Point";
+                feature["geometry"]["coordinates"]=[lng,lat];
+                var content ='<div class="popup-content"><table class="table table-striped table-bordered table-condensed">';
+                for (var i=0; i< title.length; i++) {
+                    if (title[i] != this.options.latitudeTitle && title[i] != this.options.longitudeTitle) {
+                        if(title[i]==this.options.titleForSearch){
+                            feature["properties"]["search"]=this._deleteDoubleQuotes(obj[title[i]]);
+                            feature["properties"]["title"]=this._deleteDoubleQuotes(obj[title[i]]);
+                        }else{
+                            feature["properties"][title[i]] = this._deleteDoubleQuotes(obj[title[i]]);
+                            var href='';
+                            if (obj[title[i]].indexOf('http') === 0) {
+                                href = '<a target="_blank" href="' + obj[title[i]] + '">'+ obj[title[i]] + '</a>';
+                            }
+                            if(href.length > 0)content += '<tr><th>'+title[i]+'</th><td>'+ href +'</td></tr>';
+                            else content += '<tr><th>'+title[i]+'</th><td>'+ obj[title[i]] +'</td></tr>';
+                        }
+                    }//end of if
+                }//end of for
+                content += "</table></div>";
+                feature["properties"]["popupContent"]=content;
```

Comment:
This whole popContent seems specific to your use case.
You should put every column in `properties`, and let the users decide what they do with them (html popup or whatever).